src/main/java/com/hubspot/jinjava/el/ext/AstDict.java

```diff
@@ -33,7 +33,11 @@ public Object eval(Bindings bindings, ELContext context) {
       if (entry.getKey() instanceof AstString) {
         key = Objects.toString(entry.getKey().eval(bindings, context));
       } else if (entry.getKey() instanceof AstIdentifier) {
-        key = ((AstIdentifier) entry.getKey()).getName();
+        Object result = entry.getKey().eval(bindings, context);
+        if (result == null) {
+          continue;
```

Comment:
A possible fallback here to preserve the existing behavior would be to return the identifier name when the key evaluates to `null`.

Reply:
I think the fallback is safer.